apis/dscinitialization/v1/servicemesh_types.go

```diff
@@ -0,0 +1,33 @@
+package v1
+
+import operatorv1 "github.com/openshift/api/operator/v1"
+
+// ServiceMeshSpec configures Service Mesh.
+type ServiceMeshSpec struct {
+	// +kubebuilder:validation:Enum=Managed;Removed
+	// +kubebuilder:default=Removed
+	ManagementState operatorv1.ManagementState `json:"managementState,omitempty"`
+	// Mesh holds configuration of Service Mesh used by Opendatahub.
+	Mesh MeshSpec `json:"mesh,omitempty"`
+}
+
+type MeshSpec struct {
+	// Name is a name Service Mesh Control Plane. Defaults to "minimal".
+	// +kubebuilder:default=data-science-smcp
+	Name string `json:"name,omitempty"`
+	// Namespace is a namespace where Service Mesh is deployed. Defaults to "istio-system".
+	// +kubebuilder:default=istio-system
+	Namespace string `json:"namespace,omitempty"`
+	// MetricsCollection specifies if metrics from components on the Mesh namespace
+	// should be collected. Setting the value to "Istio" will collect metrics from the
+	// control plane and any proxies on the Mesh namespace (like gateway pods). Setting
+	// to "None" will disable metrics collection.
+	// +kubebuilder:validation:Enum=Istio;None
+	// +kubebuilder:default=Istio
+	MetricsCollection string `json:"monitoring,omitempty"`
+	// IdentityType specifies the identity implementation used in the Mesh. For ROSA
+	// clusters, you would need to set this to ThirdParty.
+	// +kubebuilder:validation:Enum=Kubernetes;ThirdParty
+	// +kubebuilder:default=Kubernetes
```

Comment:
I wonder if we can't pull `maistra/api` for this purpose? https://github.com/maistra/istio-operator/blob/d922eece2343343e27016ea0d498bda36bfa64d8/pkg/apis/maistra/v2/security.go#L202-L209

One big question I have in mind is that with these kind of settings, we are drifting away from a potential support of upstream Istio and locking it to Opesnhift Service Mesh.

Reply:
This options seems to be related to this: https://istio.io/latest/docs/ops/best-practices/security/#configure-third-party-service-account-tokens.

If it is, I think it still would be applicable to upstream.

Upstream docs seem to have a hint about automating this: looks like a check against the `TokenRequest` API would tell if we can use `ThirdParty`. Perhaps we should set this to `ThirdParty` _unless_ the `TokenRequest` API is not available. From those upstream docs, it looks like `ThirdParty` should be preferred over first party. Although, I'm not sure what really would be the recommendation on Maistra.

Reply:
It is related, but wrapped in Maistra config/objects. How did you test it thus far without this feature as it is defaulting to Kubernetes (first-party-jwt)?

Reply:
I mainly use a non-ROSA cluster, so first party tokens are OK.
And... well... I haven't tested this on ROSA. I'm just trusting this change would be enough on ROSA.

Reply:
Conclusion: we will document this scenario rather than expose it as a config in DSCI. Relevant docs on OSSM side: https://access.redhat.com/documentation/en-us/openshift_container_platform/4.8/html-single/service_mesh/index#ossm-install-rosa_ossm-create-smcp

/cc @danielezonca